### PR TITLE
Build Android app on CI

### LINKED ISSUE #33: Build Android app on CI
#### Description
As an open source developer I need to make sure I won't break the Android app when committing in this project.

#### Acceptance Criteria
* Android app is built on CI
* GitHub Actions is used (`.github/workflows/android.yml`)

#### Get started

1. `clone` a fresh copy of the project or `pull` the latest `master
2. Checkout the `build_android_app_on_ci` branch: `git checkout build_android_app_on_ci`
3. Commit as many times as you need. Your commits should show up in #30  
4. When you are done or have any questions, let @platisd know.

### PR DESCRIPTION
**Pull request by platisd**
As a developer, I want to be certain that my Android app is always buildable.
Therefore, I want the app to be build after every push so I can ensure that the app is not broken.

### CODE REVIEW

File: .github/workflows/android.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+   runs-on: ubuntu-latest
+   steps:
+     - name: Set up JDK
+       uses: actions/setup-java@v1
+       with:
+         java-version: 1.8
+     - name: run Tests
+       run: |
+          cd TrailGage
+          ./gradlew test --info
```
Comment (platisd) [inline .github/workflows/android.yml]: You don't need to do that since you don't have any tests

File: .github/workflows/android.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+          
+     - name: Build App
+       run: |
+          cd TrailGage
+          ./gradlew build --stacktrace
```
Comment (platisd) [inline .github/workflows/android.yml]: Before you can `cd TrailGage` you need to clone/checkout your repository! The action will not do that automatically for you.

**Review by platisd — COMMENTED** [review]
You're very close!

**Comment by valusca** [comment]
😭😭😭😭😭😭😭😭😭😭😭😭

Hehe!

Thank you for your reply !








-------- Original message --------

File: .github/workflows/android.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+         java-version: 1.8
+     
+     - name: Clone Repo
+       uses: actions/checkout@v2
+
+      
+     - name: Build App
+       run: |
+          cd TrailGage
+          ./gradlew build --stacktrace
```
Comment (platisd) [inline .github/workflows/android.yml]: Almost done (I think)!

In your [action](https://github.com/DIT112-V20/group-11/runs/1336596228?check_suite_focus=true), you are getting the following error:

> /home/runner/work/_temp/8768e203-8951-4670-b2ab-4e0a389898e8.sh: line 2: ./gradlew: Permission denied

This is very common and the easiest solution is to make the `gradlew` binary executable. After `cd TrailGage` you can add the following command to make `gradlew` executable before running it: `chmod +x gradlew`

Also, can you please remove the blank lines between the _steps_? Unless you really like them that is. 😁

**Review by platisd — COMMENTED** [review]
Interesting. It fails because it cannot find `local.properties` which should have been automatically generated. I see others have the same problem online and I believe it has something to do with the permissions of the files you have already pushed.

> Caused by: java.io.FileNotFoundException: /home/runner/work/group-11/group-11/TrailGage/local.properties (No such file or directory)

No worries though, it seems that the solution is to just create an empty `local.properties` file before you run the `gradlew` script. Add a `touch local.properties` command after `cd TrailGage`. That should do  the trick.

File: .github/workflows/android.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+       with:
+         java-version: 1.8 
+     - name: Clone Repo
+       uses: actions/checkout@v2  
+     - name: Build App
+       run: |
+          cd TrailGage 
+          touch local.properties
+          chmod +x gradlew 
+          export TERM=xterm &&./gradlew build --stacktrace
```
Comment (platisd) [inline .github/workflows/android.yml:20]: Don't think you need the `export TERM=xterm &&` part as that's the default value in Ubuntu, but I guess it doesn't hurt either! 😁

**Review by platisd — COMMENTED** [review]
Good job! I'll send you an email soon.

**Pull request merged**